### DISCUSSION TITLE
How to get json data testing response with content: problem+json

### DISCUSSION THREAD

**Discussion by arojoal** [gavv/httpexpect#125] [Q&A] [answered]
How to get JSON data received when content type is "application/problem+json"?

I'm testing an app developed with Iris with this code:
`
result := e.POST("/pmslinkregister").WithJSON(map[string]string{"id": "test"}).Expect().Status(iris.StatusBadRequest).JSON()
`

And I get this error message:

`
expected "Content-Type" header with "application/json" media type,
        	            	but got "application/problem+json"
`
because Iris app returns "application/problem+json" with the error json. Is there a way to get the JSON with httptest when the content-type is "application/problem+json"?

I've looked for it in the code without success :-(

**Comment by gavv** [comment, marked as answer]
Hi,

Try to replace

```
.JSON()
```

with:

```
.JSON(httpexpect.ContentOpts{
    MediaType: "application/problem+json",
  })
```

**Comment by arojoal** [comment]
It worked. Thanks!!!!!!!
